test/test_restore.py

```diff
@@ -575,3 +578,29 @@ def test_restore_get_delta_basebackup_data():
         ),
     ]
     assert all(f in files for f in expected_files)
+
+
+def test_object_store_request_backup_preservation(tmp_path: Path) -> None:
```

Comment:
Wouldn't we like an additional kind of integration test with
- request backup preservation
- perform backup deletion according to the various rules
- verify the backup is there (or, even better, than it can be restored)

I mean, I see a lot of (good) unit testing, but maybe I'm missing an overall test that gives me guarantees around the preservation behaviour.

Reply:
Good point, I added tests for that. It's far from perfect, but it spans most of the code involved in backup/preservation/deletion.